src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidTestCommand.cs

```diff
@@ -210,22 +240,30 @@ protected override Task<ExitCode> InvokeInternal(ILogger logger)
 
         private string? GetDeviceToUse(ILogger logger, AdbRunner runner, string apkRequiredArchitecture)
         {
-            var allDevicesAndTheirArchitectures = runner.GetAttachedDevicesAndArchitectures();
-            if (allDevicesAndTheirArchitectures.Count > 0)
+            try
             {
-                if (allDevicesAndTheirArchitectures.Any(kvp => kvp.Value != null && kvp.Value.Equals(apkRequiredArchitecture, StringComparison.OrdinalIgnoreCase)))
+                var allDevicesAndTheirArchitectures = runner.GetAttachedDevicesAndArchitectures();
+                if (allDevicesAndTheirArchitectures.Count > 0)
```

Comment:
Imho it would be better to do `if (allDevicesAndTheirArchitectures.Count == 0)` and return fast instead of nesting since if you forget to `return` from one of the branches below, you will get an error